src/workerd/jsg/fast-api-test.c++

```diff
@@ -0,0 +1,336 @@
+#include "jsg-test.h"
+
+namespace workerd::jsg::test {
+namespace {
+
+struct WrappedInt {
+  int32_t i;
+
+  JSG_STRUCT(i);
+};
+
+class FastMethodContext: public jsg::Object, public jsg::ContextGlobal {
+ public:
+  int32_t add(int32_t a, int32_t b) {
+    return a + b;
+  }
+
+  int32_t processValue(v8::Local<v8::Value> value) {
+    KJ_ASSERT(value->IsNumber(), "Value must be a number");
+    return value.As<v8::Int32>()->Value();
+  }
+
+  // When arbitrary type unwrapping is supported, keep this test as it is.
+  int32_t processObject(v8::Local<v8::Object> obj) {
+    auto isolate = v8::Isolate::GetCurrent();
+    auto context = isolate->GetCurrentContext();
+    auto testKey = v8::String::NewFromUtf8(isolate, "test").ToLocalChecked();
+    v8::Local<v8::Value> value;
+    if (obj->Get(context, testKey).ToLocal(&value)) {
+      v8::String::Utf8Value type(isolate, value->TypeOf(isolate));
+      KJ_ASSERT(value->IsInt32(), "Received ", *type);
+      return value.As<v8::Int32>()->Value();
+    }
+    return 0;
+  }
+
+  void voidMethod(uint32_t a, uint32_t b) {
+    // Do nothing. This is testing void return type.
+  }
+
+  int32_t throwError(int32_t code) {
+    if (code > 0) {
+      JSG_FAIL_REQUIRE(TypeError, "Test error with code ", code);
+    }
+    return 0;
+  }
+
+  int32_t addWithLock(jsg::Lock& js, int32_t a, v8::Local<v8::Value> b) {
+    KJ_ASSERT(b->IsNumber(), "Second parameter must be a number");
+    int32_t bValue = b.As<v8::Int32>()->Value();
+    return a + bValue;
+  }
+
+  int32_t constAdd(int32_t a, int32_t b) const {
+    return a + b;
+  }
+
+  int32_t constAddWithLock(jsg::Lock& js, int32_t a, int32_t b) const {
+    return a + b;
+  }
+
+  int32_t unwrapStruct(jsg::Lock& js, WrappedInt w) const {
+    return w.i;
+  }
+
+  int32_t unwrapUint(jsg::Lock& js, kj::uint u) const {
+    return u;
+  }
+
+  int32_t unwrapString(jsg::Lock& js, kj::String str) const {
+    return str.size();
+  }
+
+  int32_t unwrapBufferSource(jsg::Lock& js, jsg::BufferSource source) const {
+    return source.size();
+  }
+
+  int32_t unwrapMaybe(jsg::Lock& js, kj::Maybe<kj::String> str) const {
+    KJ_IF_SOME(s, str) {
+      return s.size();
+    } else {
+      return -1;
+    }
+  }
+
+  int32_t unwrapOptional(jsg::Lock& js, jsg::Optional<kj::String> str) const {
+    KJ_IF_SOME(s, str) {
+      return s.size();
+    } else {
+      return -1;
+    }
+  }
+
+  int32_t unwrapLenientOptional(jsg::Lock& js, jsg::LenientOptional<kj::String> str) const {
+    KJ_IF_SOME(s, str) {
+      return s.size();
+    } else {
+      return -1;
+    }
+  }
+
+  JSG_RESOURCE_TYPE(FastMethodContext) {
+    JSG_TRACKABLE_FAST_METHOD(add);
```

Comment:
I'm still fairly unhappy with `JSG_TRACKABLE_FAST_METHOD` being a thing that has to exist in jsg.h given that it's only needed for tests.

Reply:
I moved this to fast-api-test.c++ file